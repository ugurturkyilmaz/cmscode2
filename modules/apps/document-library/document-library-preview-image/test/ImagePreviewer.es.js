/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {cleanup, render} from '@testing-library/react';
import React from 'react';

import ImagePreviewer from '../src/main/resources/META-INF/resources/preview/js/ImagePreviewer.es';

describe('document-library-preview-image', () => {
	afterEach(cleanup);

	it('renders an image previewer', () => {
		const {asFragment} = render(
			<ImagePreviewer
				alt="alt text"
				imageURL="image.jpg"
				spritemap="icons.svg"
			/>
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
