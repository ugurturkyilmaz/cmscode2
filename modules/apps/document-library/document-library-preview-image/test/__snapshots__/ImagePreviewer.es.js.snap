// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`document-library-preview-image renders an image previewer 1`] = `
<DocumentFragment>
  <div
    class="preview-file"
  >
    <div
      class="preview-file-container preview-file-max-height"
    >
      <img
        alt="alt text"
        class="preview-file-image"
        src="image.jpg"
        style="margin: auto auto;"
      />
    </div>
    <div
      class="preview-toolbar-container"
    >
      <div
        class="floating-bar btn-group"
        role="group"
      >
        <button
          class="btn-floating-bar btn btn-monospaced"
          title="zoom-out"
          type="button"
        >
          <svg
            class="lexicon-icon lexicon-icon-hr"
            role="presentation"
          >
            <use
              xlink:href="#hr"
            />
          </svg>
        </button>
        <button
          class="btn-floating-bar btn-floating-bar-text btn"
          title="real-size"
          type="button"
        >
          <span
            class="preview-toolbar-label-percent"
          >
            0%
          </span>
        </button>
        <button
          class="btn-floating-bar btn btn-monospaced"
          title="zoom-in"
          type="button"
        >
          <svg
            class="lexicon-icon lexicon-icon-plus"
            role="presentation"
          >
            <use
              xlink:href="#plus"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
